pennylane/templates/subroutines/out_adder.py

```diff
@@ -81,10 +81,19 @@ def __init__(
         self, x_wires, y_wires, output_wires, mod=None, work_wires=None, id=None
     ):  # pylint: disable=too-many-arguments
 
+        num_work_wires = 0 if work_wires is None else len(work_wires)
+
         if mod is None:
             mod = 2 ** (len(output_wires))
-        if (not hasattr(output_wires, "__len__")) or (mod > 2 ** len(output_wires)):
```

Comment:
For `not hasattr(output_wires, "__len__")`, is the concern that users might pass a single integer as `output_wires`? If they do and they let `mod` be the default , we've already gotten an error about `output_wires` not having a length in the check above. Maybe we could cast it to a `Wires` object if that's the concern?

Reply:
Yes, let's do that casting 👍

Reply:
I've added casting the input to `Wires` in all the templates where I've removed the `not hasattr(output_wires, "__len__")` check